src/shell/components/GlobalAccountMenu/index.tsx

```diff
@@ -44,6 +45,12 @@ export const GlobalAccountMenu: FC<GlobalAccountMenuProps> = ({
 
   const dispatch = useDispatch();
 
+  useEffect(() => {
+    if (!auth.valid) {
+      onClose();
```

Comment:
@agalin920 This makes sure that it only closes the global accounts menu once the user has been successfully logged out instead of it immediately closing and then there's a brief moment where nothing happens while waiting for the api call to resolve.